pkg/sql/sem/builtins/pg_builtins.go

```diff
@@ -454,6 +454,31 @@ func getNameForArg(ctx *eval.Context, arg tree.Datum, pgTable, pgCol string) (st
 	return string(tree.MustBeDString(r[0])), nil
 }
 
+func checkFunctionPrivilege(
```

Comment:
i think this might add additional overhead compared to how the other similar builtins are implemented with `HasAnyPrivilege`. do you know if we can use that approach?

Reply:
Oh yeah. This is a good call. I didn't know this exists. I think the approach would work. Though would need to tweak a little bit since builtin functions are not descriptor based. I'll play with it a bit. Thanks for pointing this out.